qtfred/src/ui/dialogs/ShipEditor/ShipWeaponsDialog.cpp

```diff
@@ -0,0 +1,181 @@
+	}
+}
+void ShipWeaponsDialog::on_TBLButton_clicked() {
+	if (ui->listWeapons->currentIndex().data(Qt::UserRole).toInt() >= 0) {
+		auto dialog = new WeaponsTBLViewer(this, _viewport, ui->listWeapons->currentIndex().data(Qt::UserRole).toInt());
+		dialog->show();
+	} else {
+		return;
+	}
+}
+void ShipWeaponsDialog::modeChanged(const bool enabled, const int mode)
+{
+	if (enabled) {
+		if (mode == 0) {
+			bankModel = new BankTreeModel(_model->getPrimaryBanks(), this);
+			dialogMode = 0;
+			delete weapons;
+			weapons = new WeaponModel(0);
+			ui->listWeapons->setModel(weapons);
+		} else if (mode == 1) {
+			bankModel = new BankTreeModel(_model->getSecondaryBanks(), this);
+			dialogMode = 1;
+			delete weapons;
+			weapons = new WeaponModel(1);
+			ui->listWeapons->setModel(weapons);
+		} else if (mode == 2) {
+			// bankModel = new BankTreeModel(_model->getTertiaryBanks(), this);
+			dialogMode = 2;
+		} else {
+			_viewport->dialogProvider->showButtonDialog(DialogType::Error,
+				"Illegal Mode",
+				"Somehow an Illegal mode has been set. Get a coder.\n Illegal mode is " + std::to_string(mode),
+				{DialogButton::Ok});
+			ui->radioPrimary->toggled(true);
+			bankModel = new BankTreeModel(_model->getPrimaryBanks(), this);
+			dialogMode = 0;
+		}
+		//Reconnect Meacuse the model has changed
+		connect(ui->treeBanks->selectionModel(),
+			&QItemSelectionModel::selectionChanged,
+			this,
+			&ShipWeaponsDialog::updateUI);
+		connect(ui->listWeapons->selectionModel(),
+			&QItemSelectionModel::selectionChanged,
+			this,
+			&ShipWeaponsDialog::updateUI);
+		ui->treeBanks->setModel(bankModel);
+		ui->treeBanks->expandAll();
+	}
+	updateUI();
+}
+void ShipWeaponsDialog::updateUI()
+{
+	const util::SignalBlockers blockers(this);
+	//Radio Buttons
+	ui->radioPrimary->setEnabled(!_model->getPrimaryBanks().empty());
+	ui->radioSecondary->setEnabled(!_model->getSecondaryBanks().empty());
+	ui->radioTertiary->setEnabled(false);
+
+	ui->treeBanks->expandAll();
+	// Setall button
+	if (ui->treeBanks->getTypeSelected() == 0) {
+		ui->setAllButton->setEnabled(true);
+	} else {
+		ui->setAllButton->setEnabled(false);
+	}
+	// Change AI Button
+	if (ui->treeBanks->getTypeSelected() == 1) {
+		ui->AIButton->setEnabled(true);
+	} else {
+		ui->AIButton->setEnabled(false);
+	}
+	// AI Combo Box
+	ui->AICombo->clear();
+	for (int i = 0; i < Num_ai_classes; i++) {
+		ui->AICombo->addItem(Ai_class_names[i], QVariant(i));
+	}
+	ui->AICombo->setCurrentIndex(ui->AICombo->findData(m_currentAI));
+	if (ui->listWeapons->selectionModel()->hasSelection() &&
+		ui->listWeapons->currentIndex().data(Qt::UserRole).toInt() != -1) {
+		ui->TBLButton->setEnabled(true);
+	} else {
+		ui->TBLButton->setEnabled(false);
+	}
+}
+
+void ShipWeaponsDialog::aiClassChanged(const int index) {
+	m_currentAI = ui->AICombo->itemData(index).toInt();
+}
+
+void ShipWeaponsDialog::on_AIButton_clicked() {
```

Comment:
It'd be good to follow a naming pattern throughout QtFRED if we can. This button should probably be aiButton. Minor nitpick, really, but the consistency does help overall later on.